### PR TITLE
Don't run deployment tests during PR CI (it's already sufficient that they are constantly running on dev) + refactoring

### PR DESCRIPTION
**Pull request by lbeder**
- Don't run deployment tests during PR CI (it's already sufficient that they are constantly running on dev)
- Split unit-tests and deployment tests into two separate jobs during the nightly run

### CODE REVIEW

**Review by barakman — APPROVED** [review]

**Review by yudilevi — APPROVED** [review]

**Review by yudilevi — DISMISSED** [review]

File: deployments/setup-fork.ts [hunk 1]
```diff
@@ -31,13 +31,14 @@ const {
...
 const tenderlyNetwork = tenderly.network();
+let forkId: string;
 
 const createTenderlyFork = async () => {
     console.log('Setting up new fork...');
 
     await tenderlyNetwork.initializeFork();
 
-    const forkId = tenderlyNetwork.getFork()!;
+    forkId = tenderlyNetwork.getFork()!;
```
Comment (barakman) [inline deployments/setup-fork.ts:41]: This change looks redundant.

File: utils/Deploy.ts [hunk 2]
```diff
@@ -229,7 +229,7 @@ const saveTypes = async (options: SaveTypeOptions) => {
     const { name, contract } = options;
 
     // don't attempt to save the types for legacy contracts
-    if (Object.keys(LegacyInstanceName).includes(name)) {
+    if (Object.keys(LegacyInstanceNameV2).includes(name) || Object.keys(LegacyInstanceName).includes(name)) {
```
Comment (barakman) [inline utils/Deploy.ts:232]: Minor - can use:
```
if (Object.keys({...LegacyInstanceNameV2, ...LegacyInstanceName}).includes(name))
```

**Review by barakman — APPROVED** [review]

File: deployments/setup-fork.ts (hunk 1 above)
Comment (lbeder) [reply deployments/setup-fork.ts:41]: Why? It's setting the new global `forkId` variable

File: utils/Deploy.ts (hunk 2 above)
Comment (lbeder) [reply utils/Deploy.ts]: Refactored differently. Please take a look

**Pull request merged**